sdk/identity/identity/samples/AzureCacheForRedis/ioredis.md

```diff
@@ -0,0 +1,273 @@
+
+* [Client Certificate Credential](https://docs.microsoft.com/javascript/api/@azure/identity/clientcertificatecredential?view=azure-node-latest)
+* [Client Secret Credential](https://docs.microsoft.com/javascript/api/@azure/identity/clientsecretcredential?view=azure-node-latest)
+* [Managed Identity Credential](https://docs.microsoft.com/javascript/api/@azure/identity/managedidentitycredential?view=azure-node-latest)
+* [Username Password Credential](https://docs.microsoft.com/javascript/api/@azure/identity/usernamepasswordcredential?view=azure-node-latest)
+* [Azure CLI Credential](https://docs.microsoft.com/javascript/api/@azure/identity/azureclicredential?view=azure-node-latest)
+* [Interactive Browser Credential](https://docs.microsoft.com/javascript/api/@azure/identity/interactivebrowsercredential?view=azure-node-latest)
+* [Device Code Credential](https://docs.microsoft.com/javascript/api/@azure/identity/devicecodecredential?view=azure-node-latest)
+
+#### Authenticate with Azure AD: Handle Reauthentication
+
+This sample is intended to assist in authenticating a hosted Azure Cache for Redis instance with Azure AD via ioredis. It focuses on displaying the logic required to fetch an Azure AD access token and to use it as the password when setting up the ioredis instance. It further shows how to recreate and authenticate the ioredis instance when its connection is broken in error/exception scenarios.
+
+##### Migration Guidance
+
+When migrating your existing application code to authenticate with Azure AD, replace the password input with Azure AD token.
+Integrate the logic in your application code to fetch an Azure AD access token via the Azure Identity library, as shown below. Replace the password configuring/retrieving logic in your application code.
+
+```ts
+import Redis from "ioredis";
+import { DefaultAzureCredential, TokenCredential } from "@azure/identity";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+async function returnPassword(credential: TokenCredential) {
+  try {
+    // The scope will be changed for Azure AD Public Preview
+    const redisScope = "https://*.cacheinfra.windows.net:10225/appid/.default"
+
+    // Fetch an Azure AD token to be used for authentication. This token will be used as the password.
+    let accessTokenObject = await credential.getToken(redisScope);
+    return accessTokenObject;
+  } catch (e) {
+    throw e;
+  }
+}
+
+async function main() {
+  // Construct a Token Credential from Identity library
+  const credential = new DefaultAzureCredential();
+  let accessTokenObject = await returnPassword(credential);
```

Comment:
`token` or `accessToken`. `accessTokenObject` is a bit excessive.

Reply:
I want to be specific that they should use the `token` field of this object as the password rather than the object directly. Using `accessTokn.token` to me looked repititive

Reply:
fair enough